subsys/bluetooth/audio/csip_crypto.c

```diff
@@ -81,12 +81,9 @@ int bt_csip_sih(const uint8_t sirk[BT_CSIP_SET_SIRK_SIZE], uint32_t r,
 
 	LOG_DBG("BE: r' %s", bt_hex(res, sizeof(res)));
 
-	if (IS_ENABLED(CONFIG_LITTLE_ENDIAN)) {
-		/* Swap to Big Endian (BE) */
-		sys_memcpy_swap(sirk_tmp, sirk, BT_CSIP_SET_SIRK_SIZE);
-	} else {
-		(void)memcpy(sirk_tmp, sirk, BT_CSIP_SET_SIRK_SIZE);
-	}
+	/* Swap to Big Endian (BE) */
+	sys_memcpy_swap(sirk_tmp, sirk, BT_CSIP_SET_SIRK_SIZE);
+
```

Comment:
Well, that's true. The SIRK is an array of 16 uint8 per spec.
But it is still relevant to swap, I guess because the crypto functions are requiring it to be in that order - in relation to how it's read by the client.

Reply:
If the SIRK is an array of bytes, then the order of how it is read should be the same regardless of host endianess, right? 

We swap here because we are using the `bt_encrypt_be`, because the underlying encryption functions are BIG endianess, and IIRC they do not consider they key (in this case the SIRK) 16 1-byte values, but rather a single 16-byte value, so this is where endianess starts to matter. 

At this point I don't recall why we are using BE here in the first place. 

But in any case, the comment doesn't really make sense in its current form, but should perhaps be something like
```
Swap to a 128-bit BE key for bt_encrypt_be
```